src/main/kotlin/controllers/LocationSearchController.kt

```diff
@@ -0,0 +1,147 @@
+package controllers
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.google.inject.Inject
+import javafx.beans.value.ObservableValue
+import javafx.collections.FXCollections
+import javafx.collections.transformation.FilteredList
+import javafx.fxml.FXML
+import javafx.scene.control.Button
+import javafx.scene.control.Label
+import javafx.scene.control.ListCell
+import javafx.scene.control.ListView
+import javafx.scene.control.TextField
+import javafx.scene.control.TextFormatter
+import javafx.scene.input.KeyCode
+import javafx.scene.input.KeyEvent
+import javafx.scene.layout.AnchorPane
+import javafx.scene.text.TextAlignment
+import javafx.stage.Stage
+import javafx.util.Callback
+import models.locations.Location
+import models.locations.Locations
+import models.scene.Scene
+import utils.Utils
+import java.util.regex.Pattern
+
+class LocationSearchController @Inject constructor(
+    private val scene: Scene
+) {
+    @FXML
+    lateinit var wrapper: AnchorPane
+
+    @FXML
+    lateinit var txtSearchQuery: TextField
+
+    @FXML
+    lateinit var listLocations: ListView<Location>
+
+    @FXML
+    lateinit var txtRadius: TextField
+
+    @FXML
+    private lateinit var btnLoad: Button
+
+    @FXML
+    private lateinit var lblErrorText: Label
+
+    private var selectedLocation: Location? = null
+
+    @FXML
+    private fun handleKeyPressed(e: KeyEvent) {
+        if (e.code == KeyCode.ENTER)
+            btnLoad.fire()
+    }
+
+    @FXML
+    private fun initialize() {
+        listLocations.cellFactory = LocationCellFactory()
```

Comment:
Does Kotlin allow for you to implement this interface as an inline closure instead?